subsys/CMakeLists.txt

```diff
@@ -1,3 +1,4 @@
+add_subdirectory(app_memory)
```

Comment:
I guess we can make this into add_subdirectory_ifdef(CONFIG_APP_SHARED_MEM, app_memory)